lib/commands/simctl.js

```diff
@@ -0,0 +1,20 @@
+const SIMCTL_FEATURE = 'simctl';
+
+const commands = {
+  /**
+   * Run the given command with arguments as `xcrun simctl` subcommand.
+   * This method works behind the 'simctl' security flag.
+   * @param {string} command Subcommand to run with `xcrun simctl`
+   * @param {Array} [args=[]]
+   * @returns {Promise<Record<'stdout' | 'stderr' | 'code', number>>} The output of 'exec'.
+   */
+  async mobileSimctl(command, args = []) {
+    this.ensureFeatureEnabled(SIMCTL_FEATURE);
```

Comment:
theoretically we could add it without the feature flag if we only limit available args to the current device.

In general I am not really convinced on why someone would need to access other simulators during the test

Reply:
I'm not familiar with subcommands, which can be limited only to the current device. Maybe we should explore available subcommands and limit available subcommands which can target to the current device...? Then, it could take time, so will close this PR and would like to wait for someone who is interested in adding a new subcommand rather than completing this PR. (i don't have a strong motivation for this, so it's fine)

Reply:
you may check `xcrun simctl help` for the full list of commands. 

Do you need my help to create a whitelist of commands that require device udid to work?

Reply:
Oh, yea. if you already explored subcommands, it will help

Reply:
These are all commands:

```
	addmedia            Add photos, live photos, videos, or contacts to the library of a device.
	boot                Boot a device or device pair.
	clone               Clone an existing device.
	create              Create a new device.
	delete              Delete specified devices, unavailable devices, or all devices.
	diagnose            Collect diagnostic information and logs.
	erase               Erase a device's contents and settings.
	get_app_container   Print the path of the installed app's container
	getenv              Print an environment variable from a running device.
	help                Prints the usage for a given subcommand.
	icloud_sync         Trigger iCloud sync on a device.
	install             Install an app on a device.
	install_app_data    Install an xcappdata package to a device, replacing the current contents of the container.
	io                  Set up a device IO operation.
	keychain            Manipulate a device's keychain
	launch              Launch an application by identifier on a device.
	list                List available devices, device types, runtimes, or device pairs.
	location            Control a device's simulated location
	logverbose          enable or disable verbose logging for a device
	openurl             Open a URL in a device.
	pair                Create a new watch and phone pair.
	pair_activate       Set a given pair as active.
	pbcopy              Copy standard input onto the device pasteboard.
	pbpaste             Print the contents of the device's pasteboard to standard output.
	pbsync              Sync the pasteboard content from one pasteboard to another.
	privacy             Grant, revoke, or reset privacy and permissions
	push                Send a simulated push notification
	rename              Rename a device.
	runtime             Perform operations on runtimes
	shutdown            Shutdown a device.
	spawn               Spawn a process by executing a given executable on a device.
	status_bar          Set or clear status bar overrides
	terminate           Terminate an application by identifier on a device.
	ui                  Get or Set UI options
	uninstall           Uninstall an app from a device.
	unpair              Unpair a watch and phone pair.
	upgrade             Upgrade a device to a newer runtime.
```

the whitelist should only include the following commands:

```
addmedia
get_app_container
getenv
icloud_sync
install
install_app_data
io
keychain
launch
location
logverbose
openurl
pbcopy
pbpaste
pbsync
privacy
push
shutdown
spawn
status_bar
terminate
ui
uninstall
```

Reply:
for each command we must also make sure the UUID of the device is the same as the one that has been chosen during session creation

Reply:
thank you